File: InternetForumBackend/API/Data/Models/PostVote.cs
namespace API.Data.Models;

public class PostVote
{
    public int VoteId { get; set; }
    public int VoteCount { get; set; }

    public int PostId { get; set; }

    public Post Post { get; set; }

    public ICollection<UserPostVote> UserPostVotes { get; set; }

    public PostVote()
    {
        VoteCount = 0;
    }
}

File: InternetForumFrontend/src/app/Models/PostVote.ts
import { Post } from "./Post";
import { UserPostVote } from "./UserPostVote";

// NOTE: ? = null, ! = not null
export class PostVote{ 
        voteId?: number;
        voteCount!: number;
        postId?: number;
        post?: Post;
        userPostVotes?: UserPostVote;
}